src/main/java/org/truffleruby/core/hash/HashLiteralNode.java

```diff
@@ -28,13 +29,21 @@ protected HashLiteralNode(RubyNode[] keyValues) {
         this.keyValues = keyValues;
     }
 
+    private static final boolean bigHashTypeIsCompactHash;
+    static {
+        String type = System.getProperty("BigHashStrategy");
+        bigHashTypeIsCompactHash = type != null && type.equalsIgnoreCase("compact");
```

Comment:
This option affects new Hash literals but does *not* affect packed hashes becoming too big to be packed, those still transition to the buckets strategy. You need to change this in `org.truffleruby.core.hash.library.PackedHashStoreLibrary#promoteToBuckets` or its only caller, so if the option is set then it promotes to CompactHashStore instead.